test/test_util.py

```diff
@@ -95,6 +96,8 @@ def test_invalid_host(self):
             'http://google.com:foo',
             'http://::1/',
             'http://::1:80/',
+            'http://google.com:-80',
+            six.u('http://google.com:\xb2\xb2'),  # \xb2 = ^2
```

Comment:
Surrounding whitespace is acceptable: everything else is not. Per [RFC 7231 Section 7.1.3](https://tools.ietf.org/html/rfc7231#section-7.1.3), the value is "is a non-negative decimal integer, representing time in seconds."